### PR TITLE
Update to .Net 7 and updated NuGet Dependencies

### PR DESCRIPTION
**Pull request by MartinBelton-gov**
Update to .Net 7 and updated NuGet Dependencies

### CODE REVIEW

File: src/FamilyHub.IdentityServerHost/Persistence/Repository/ApplicationDbContextInitialiser.cs [hunk 1]
```diff
@@ -115,7 +102,7 @@ private async Task EnsureRoles()
                 IdentityResult result = await _roleManager.CreateAsync(new IdentityRole(role));
                 if (!result.Succeeded)
                 {
-                    throw new Exception(result.Errors.First().Description);
+                    throw new ArgumentException(result.Errors.First().Description);
```
Comment (hardik-desai-kcl) [inline src/FamilyHub.IdentityServerHost/Persistence/Repository/ApplicationDbContextInitialiser.cs:105]: clearly this is not an argument exception, invalid operation exception perhaps?

File: src/FamilyHub.IdentityServerHost/Persistence/Repository/ApplicationDbContextInitialiser.cs [hunk 2]
```diff
@@ -195,7 +181,7 @@ private async Task AddUser(UserManager<ApplicationIdentityUser> userMgr, string
             var result = userMgr.CreateAsync(user, password).Result;
             if (!result.Succeeded)
             {
-                throw new Exception(result.Errors.First().Description);
+                throw new ArgumentException(result.Errors.First().Description);
```
Comment (hardik-desai-kcl) [inline src/FamilyHub.IdentityServerHost/Persistence/Repository/ApplicationDbContextInitialiser.cs:184]: not an ArgumentException

File: src/FamilyHub.IdentityServerHost/Persistence/Repository/ApplicationDbContextInitialiser.cs [hunk 3]
```diff
@@ -206,21 +192,21 @@ private async Task AddUser(UserManager<ApplicationIdentityUser> userMgr, string
...
             }
 
             result = await userMgr.AddToRoleAsync(user, role);
             if (!result.Succeeded)
             {
-                throw new Exception(result.Errors.First().Description);
+                throw new ArgumentException(result.Errors.First().Description);
```
Comment (hardik-desai-kcl) [inline src/FamilyHub.IdentityServerHost/Persistence/Repository/ApplicationDbContextInitialiser.cs:201]: not an ArgumentException

File: src/FamilyHub.IdentityServerHost/Persistence/Repository/ApplicationDbContextInitialiser.cs [hunk 4]
```diff
@@ -138,8 +125,8 @@ private async Task EnsureUsers()
 
         if (_isProduction)
         {
-            await AddUser(_userManager, "Ben Machinnes", "Ben.MACINNES@education.gov.uk", "8qBoCcDrvP$", "DfEAdmin", "www.warmhandover.gov.uk");
-            await AddUser(_userManager, "Emily Chomitzki@digital", "emily.chomitzki@digital.education.gov.uk", "gXT8lp9qOW$", "DfEAdmin", "www.warmhandover.gov.uk");
+            await AddUser(_userManager, "Ben Machinnes", "Ben.MACINNES@education.gov.uk", "8qBoCcDrvP$", "DfEAdmin");
```
Comment (hardik-desai-kcl) [inline src/FamilyHub.IdentityServerHost/Persistence/Repository/ApplicationDbContextInitialiser.cs:128]: this is visible to whole of the world as the code is open source, is this appropriate?

File: tests/FamilyHub.IdentityServerHost.UI.UnitTests/MockHelpers.cs [hunk 5]
```diff
@@ -45,7 +45,7 @@ public static class MockHelpers
         pwdValidators.Add(new PasswordValidator<TUser>());
         var userManager = new UserManager<TUser>(store, options.Object, new PasswordHasher<TUser>(),
             userValidators, pwdValidators, MockLookupNormalizer(),
-            new IdentityErrorDescriber(), null,
+            new IdentityErrorDescriber(), default!,
```
Comment (hardik-desai-kcl) [inline tests/FamilyHub.IdentityServerHost.UI.UnitTests/MockHelpers.cs:48]: default! says this value is default and not null, however previous value was null, either change this to spcific value i.e. new IServiceProvider services, or remove ! to match previous behaviour?

File: tests/FamilyHub.IdentityServerHost.UI.UnitTests/MockHelpers.cs [hunk 6]
```diff
@@ -60,7 +60,7 @@ public static class MockHelpers
         return new RoleManager<TRole>(store, roles,
             MockLookupNormalizer(),
             new IdentityErrorDescriber(),
-            null);
+            default!);
```
Comment (hardik-desai-kcl) [inline tests/FamilyHub.IdentityServerHost.UI.UnitTests/MockHelpers.cs:63]: same as above

**Review by hardik-desai-kcl — CHANGES_REQUESTED** [review]

File: src/FamilyHub.IdentityServerHost/Api/Controllers/AuthenticateController.cs [hunk 7]
```diff
@@ -40,15 +40,18 @@ public class AuthenticateController : ControllerBase
     [Route("login")]
     public async Task<IActionResult> Login([FromBody] ApiLoginModel model)
     {
-        var user = await _userManager.FindByNameAsync(model.Username);
-        if (user != null && await _userManager.CheckPasswordAsync(user, model.Password))
+        if (model == null)
```
[the remark this answers is not in the document]
Comment (MartinBelton-gov) [reply src/FamilyHub.IdentityServerHost/Api/Controllers/AuthenticateController.cs:43]: Done

File: src/FamilyHub.IdentityServerHost/Api/Controllers/AuthenticateController.cs [hunk 8]
```diff
@@ -40,15 +40,18 @@ public class AuthenticateController : ControllerBase
...
+            return Unauthorized();
+
+        var user = await _userManager.FindByNameAsync(model.Username ?? string.Empty);
```
[the remark this answers is not in the document]
Comment (MartinBelton-gov) [reply src/FamilyHub.IdentityServerHost/Api/Controllers/AuthenticateController.cs:46]: Done

File: src/FamilyHub.IdentityServerHost/Areas/Gds/Pages/Invitation/CheckAccountDetails.cshtml.cs [hunk 9]
```diff
@@ -33,20 +33,14 @@ public void OnGet()
         RoleName = RoleHelper.GetRoleFullName(NewUser?.Role ?? string.Empty);
     }
 
+#pragma warning disable S4457
```
[the remark this answers is not in the document]
Comment (MartinBelton-gov) [reply src/FamilyHub.IdentityServerHost/Areas/Gds/Pages/Invitation/CheckAccountDetails.cshtml.cs:36]: Remove #pragma

File: src/FamilyHub.IdentityServerHost/Api/Controllers/AuthenticateController.cs [hunk 10]
```diff
@@ -162,10 +146,12 @@ public async Task<IActionResult> RegisterAdmin([FromBody] ApiRegisterModel model
     }
 #endif
 
+#pragma warning disable S4457
     [HttpPost]
     [Route("refresh-token")]
     public async Task<IActionResult> RefreshToken(TokenModel tokenModel)
     {
+
         if (tokenModel is null)
```
[the remark this answers is not in the document]
Comment (MartinBelton-gov) [reply src/FamilyHub.IdentityServerHost/Api/Controllers/AuthenticateController.cs:155]: Done

File: src/FamilyHub.IdentityServerHost/Areas/Gds/Pages/Invitation/WhichOrganisation.cshtml.cs [hunk 11]
```diff
@@ -82,19 +82,24 @@ private async Task Init()
...
         {
             var userEmail = User.FindFirstValue(ClaimTypes.Email);
+            ArgumentNullException.ThrowIfNull(userEmail);
             var user = await _userManager.FindByEmailAsync(userEmail);
             var orgaisationIds = _organisationRepository.GetAllUserOrganisationsByUserId(user.Id);
             OrganisationSelectionList = list.Where(x => orgaisationIds.Contains(x.Id)).OrderBy(x => x.Name).Select(c => new SelectListItem() { Text = c.Name, Value = c.Id }).ToList();
         }
         else if (User.IsInRole("LAAdmin"))
         {
             var userEmail = User.FindFirstValue(ClaimTypes.Email);
```
[the remark this answers is not in the document]
Comment (MartinBelton-gov) [reply src/FamilyHub.IdentityServerHost/Areas/Gds/Pages/Invitation/WhichOrganisation.cshtml.cs:92]: Fixed

File: src/FamilyHub.IdentityServerHost/Persistence/Repository/ApplicationDbContextInitialiser.cs (hunk 3 above)
Comment (MartinBelton-gov) [reply src/FamilyHub.IdentityServerHost/Persistence/Repository/ApplicationDbContextInitialiser.cs:201]: Fixed

File: src/FamilyHub.IdentityServerHost/Persistence/Repository/ApplicationDbContextInitialiser.cs (hunk 4 above)
Comment (MartinBelton-gov) [reply src/FamilyHub.IdentityServerHost/Persistence/Repository/ApplicationDbContextInitialiser.cs:128]: Will raise a ticket

File: tests/FamilyHub.IdentityServerHost.UI.UnitTests/MockHelpers.cs (hunk 5 above)
Comment (MartinBelton-gov) [reply tests/FamilyHub.IdentityServerHost.UI.UnitTests/MockHelpers.cs:48]: Fixed

**Review by hardik-desai-kcl — APPROVED** [review]

**Pull request merged**